### ISSUE TITLE
Provide specific options for configuring the number of io and worker HTTP threads

### ISSUE THREAD

**Issue by pedroigor** [keycloak/keycloak#15224]
### Description

Currently, we rely on Quarkus options to configure the io and worker thread HTTP pools.

Although the defaults should be enough for the majority of use cases, ideally we should provide specific options on our side as this is a very common setting when tunning the server.

### Motivation

The server should provide good defaults for the majority of the use cases. Most of the time, users don't need to manually set the size for the io and thread pools.

The threads from the IO thread pool are responsible to handle incoming request events and dispatch those events to worker threads. Its size has a direct impact on how many requests the server can serve and how fast it will dispatch to the worker thread pool.

On the other hand, the worker thread pool is responsible for actually performing the heavy work and its size
has a direct impact on how many concurrent clients you can serve in parallel.

Users should be able to tune the server accordingly to their needs if the defaults are not enough to support their performance requirements. Because of that, we should not rely on any setting from the underlying stack (e.g.: quarkus or undertow) but provide a specific Keycloak option to set these options. 

Perhaps, this issue should also include some benchmarking to make sure how good our defaults are when running on constrained environments like containers.

**Comment by stianst** [comment]
Prior to adding these options we should have an understanding of when to use these options and some recommendations on what to change the numbers to. We should probably also perform some bench-marking to validate there's actual improvements.

I can imagine there's two cases where the defaults would not be ideal:

* Very small nodes: the resources allocated to a node is too small to handle the default size, and either throughput is increased significantly by lowering the numbers, or the throughput remains the same while memory consumption being decreased significantly
* Very big nodes: the default size doesn't take advantage of all the CPU or memory allocated to the nodes, and throughout can be increased by increase the numbers

**Comment by pedroigor** [comment]
I agree it should also include a tunning guide. It should probably be an outcome of benchmarks using our benchmark tool.

The default value for the io/event-loop thread pool should be safe/good enough for most of the uses. After Quarkus 2.13, the default is set to match the number of available processors. For constrained environments that sounds reasonable to me although you might still adjust it to either lower it to reduce memory footprint or, if proven, keep/raise it for better throughput.

What concerns me most is the default size for the worker pool, which by default is set to a maximum of `200` if the formula `8 * processors` is lower than `200`. This one might have a negative impact depending on how many cpus you have available due to thread contention.

**Comment by vmuzikar** [comment]
@pedroigor @ahus1 Do you think this issue was superseded by #24684 and #26455?

**Comment by pedroigor** [comment]
@vmuzikar I think so.
